dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/dispatch/host/LowerWeightHostManager.java

```diff
@@ -154,6 +155,11 @@ public void run() {
         }
 
         public HostWeight getHostWeight(String addr, String workerGroup, String heartBeatInfo) {
+            if (StringUtils.isEmpty(heartBeatInfo)) {
```

Comment:
Maybe we can change the return type of `org.apache.dolphinscheduler.server.master.registry.ServerNodeManager#getWorkerNodeInfo(java.lang.String)` to `Optional<String>` and check whether it's `Optional.isPresent()`?

Reply:
@kezhenxu94 actually yes, and this pr is for relove this issue 
[Bug] [MasterServer] RefreshResourceTask error by NPE #6641

and i will open an new issue for optimize and improve the rubustness of the interface
and pull a sepearte request，

Reply:
I agree zhenxu, use optional is better. If you think it is unnecessary to change `ServerNodeManager#getWorkerNodeInfo` in this pr, maybe it's better to change the return type of `LowerWeightHostManager.RefreshResourceTask#getHostWeight` to return  `Optional<HostWeight>`? This is a small change.

Reply:
@ruanwenjun @kezhenxu94 yes, it a elegent way to deal the npe with optional,

and now we are better to fouce the issue releated area code, and later will consider the whole